pkg/virt-controller/watch/pool.go

```diff
@@ -605,30 +621,102 @@ func injectHashIntoVM(vm *virtv1.VirtualMachine, vmHash string, vmiHash string)
 		vm.Spec.Template.ObjectMeta.Labels = map[string]string{}
 	}
 
-	vm.Labels[virtv1.VirtualMachineTemplateHash] = vmHash
-	vm.Spec.Template.ObjectMeta.Labels[virtv1.VirtualMachineInstanceTemplateHash] = vmiHash
+	vm.Labels[virtv1.VirtualMachinePoolRevisionName] = revisionName
+	vm.Spec.Template.ObjectMeta.Labels[virtv1.VirtualMachinePoolRevisionName] = revisionName
 
 	return vm
 }
 
+func getRevisionName(pool *poolv1.VirtualMachinePool) string {
+	return fmt.Sprintf("%s-%d", pool.Name, pool.Generation)
+}
+
+func (c *PoolController) createControllerRevision(pool *poolv1.VirtualMachinePool) (string, error) {
+	poolKey, err := controller.KeyFunc(pool)
+	if err != nil {
+		return "", err
+	}
+	revisionName := getRevisionName(pool)
+
+	_, alreadyExists, err := c.getControllerRevision(pool.Namespace, revisionName)
+	if err != nil {
+		return "", err
+	} else if alreadyExists {
+		// already created
+		return revisionName, nil
```

Comment:
Maybe call this function `ensureControllerRevision` instead of `createControllerRevision`.

Reply:
fixed